project/src/main/java/com/google/sps/database/JDBCLib.java

```diff
@@ -1,155 +1,123 @@
-    public void update(Report report) {
+    public void update(Report report) throws SQLException{
         String entryID = report.entryID;
         String Query = "UPDATE collisionReports\n" +
                 "SET title = \"" + report.title + "\", latitude = \"" + report.latitude + "\", longitude = \"" +
-                report.latitude + "\", reportDate = \"" + report.date + "\", reportDescription = \"" +
-                report.description + "\", contactDetails = \"" + report.contactDetails + "\", imageURL = \"" +
+                report.longitude + "\", reportDescription = \"" + report.description +
+                "\", reportDate = \"" + report.date +
+                "\", contactDetails = \"" + report.contactDetails + "\", imageURL = \"" +
                 report.imageURL + "\"\n" +
                 "WHERE entryID = " + entryID + ";";
-        try {
 
-            Connection connection = DriverManager.getConnection(url,
-                    user, password);
 
-            Statement statement = connection.createStatement();
+        Connection connection = DriverManager.getConnection(url,
+                user, password);
 
-            statement.executeUpdate(Query);
-        } catch (Exception e) {
-            e.printStackTrace();
+        Statement statement = connection.createStatement();
+
+        statement.executeUpdate(Query);
 
-        }
     }
 
-    //
-    public Report getEntry(String entryID) {
-        String Query = "SELECT * FROM collisionReports where entryID = " + entryID;
 
-        try {
-
-            Connection connection = DriverManager.getConnection(url,
-                    user, password);
+    public Report getReport(String entryID) throws SQLException {
+        String Query = "SELECT * FROM collisionReports where entryID = " + entryID;
 
-            Statement statement = connection.createStatement();
 
-            ResultSet resultSet = statement.executeQuery(Query);
+        Connection connection = DriverManager.getConnection(url,
+                user, password);
 
-            while (resultSet.next()) {
-                String title = resultSet.getString("title");
-                String latitude = resultSet.getString("latitude");
-                String longitude = resultSet.getString("longitude");
-                String reportDate = resultSet.getString("reportDate");
-                String reportDescription = resultSet.getString("reportDescription");
-                String contactDetails = resultSet.getString("contactDetails");
-                String imageURL = resultSet.getString("imageURL");
+        Statement statement = connection.createStatement();
 
-                Report report = new Report(title,latitude,longitude,
-                        reportDate,reportDescription,contactDetails,imageURL,entryID);
+        ResultSet resultSet = statement.executeQuery(Query);
 
-                return report;
-            }
+        while (resultSet.next()) {
```

Comment:
Instead of iterating over the resultSet (which should be a single result), let's first check that the resultSet is actually just one result and then just retrieve it.